packages/neuron-wallet/src/services/addresses.ts

```diff
@@ -149,128 +160,196 @@ export default class AddressService {
-    ).address
-
-    const mainnetAddress: string = addressMetaInfo.accountExtendedPublicKey.address(
+    const address: string = addressMetaInfo.accountExtendedPublicKey.address(
       addressMetaInfo.addressType,
       addressMetaInfo.addressIndex,
-      AddressPrefix.Mainnet
+      AddressService.getAddressPrefix()
     ).address
 
-    const addressToParse = NetworksService.getInstance().isMainnet() ? mainnetAddress : testnetAddress
-    const blake160: string = AddressParser.toBlake160(addressToParse)
+    const blake160: string = AddressParser.toBlake160(address)
 
-    const testnetAddressInfo: AddressInterface = {
+    const addressInfo: AddressInterface = {
       walletId: addressMetaInfo.walletId,
-      address: testnetAddress,
+      address,
       path,
       addressType: addressMetaInfo.addressType,
       addressIndex: addressMetaInfo.addressIndex,
-      txCount: 0,
-      liveBalance: '0',
-      sentBalance: '0',
-      pendingBalance: '0',
-      balance: '0',
-      blake160,
-      version: AddressVersion.Testnet,
+      blake160
     }
 
-    const mainnetAddressInfo = {
-      ...testnetAddressInfo,
-      address: mainnetAddress,
-      version: AddressVersion.Mainnet,
-    }
-
-    return [testnetAddressInfo, mainnetAddressInfo]
+    return addressInfo
   }
 
-  public static nextUnusedAddress = (walletId: string): AddressInterface | undefined => {
-    const addressEntity = AddressDao.nextUnusedAddress(walletId,  AddressService.getAddressVersion())
-    if (!addressEntity) {
+  private static async maxAddressIndex(walletId: string, addressType: AddressType): Promise<number | undefined> {
+    const result = await getConnection()
+      .getRepository(HdPublicKeyInfo)
+      .createQueryBuilder()
+      .select('addressIndex')
+      .where({walletId, addressType})
+      .orderBy('addressIndex', 'DESC')
+      .getRawOne()
+
+    if (!result) {
       return undefined
     }
-    return addressEntity
+
+    return result.addressIndex
   }
 
-  public static allUnusedReceivingAddresses(walletId: string): AddressInterface[] {
-    return AddressDao.allUnusedReceivingAddresses(walletId,  AddressService.getAddressVersion())
+  private static async getUnusedAddresses(walletId: string) {
+    const allUnusedAddresses = await this.getAllUnusedAddressesByWalletId(walletId)
+
+    const unusedReceivingAddresses = allUnusedAddresses
+      .filter(addr => addr.addressType === AddressType.Receiving)
+    const unusedChangeAddresses = allUnusedAddresses
+      .filter(addr => addr.addressType === AddressType.Change)
+
+    return [unusedReceivingAddresses, unusedChangeAddresses]
   }
 
-  public static nextUnusedChangeAddress = (walletId: string): AddressInterface | undefined => {
-    const addressEntity = AddressDao.nextUnusedChangeAddress(walletId,  AddressService.getAddressVersion())
-    if (!addressEntity) {
-      return undefined
+  public static async getAllUnusedAddressesByWalletId (walletId: string) {
```

Comment:
Hard to tell the difference between `getUnusedAddresses` and `getAllUnusedAddressesByWalletId` by their names.